### DISCUSSION TITLE
How to get all users and group with permissions on a list/library

### DISCUSSION THREAD

**Discussion by SP-Support** [pnp/powershell#580] [Q&A]
How do you get a list of users/groups who have permission on a list/library?

I would thing that `Get-PnPListPermissions` would have an option of just showing a list of users but you have to use a principalID.

**Comment by ToddKlindt** [comment]
This is a tough one. Permissions are a property of each individual object. Because of sharing any file or folder can have unique permissions. To get the report you're looking for you have to look at every object in the list/library, see if it's inheriting permissions, and if not get its unique permissions. I haven't run it myself, but at first glance it looks like this script might get you want you're looking for, https://www.sharepointdiary.com/2019/02/sharepoint-online-pnp-powershell-to-export-document-library-permissions.html
